### ISSUE TITLE
Error: Write to EEPROM failed @ 0x0e

### ISSUE THREAD

**Issue by Gowresh-MARC** [eblot/pyftdi#214]
Hi guys,
 I am running on ubuntumate on raspberry pi .

I tried the following command to change the product name of my ftdi:-

`/pyftdi/pyftdi/bin$ ftconf.py ftdi://ftdi:232:A50285BI/1 -p NANOU1 -u`

I got the following error:-
Error: Write to EEPROM failed @ 0x0e

Note: I unloaded the usbserial and ftdi-sio modules

**Comment by len0rd** [comment]
tl;dr, I have [a branch you could try](https://github.com/len0rd/pyftdi/tree/feature/len0rd/eeprom_mirror), that may or may not fix your problem


Have you tried dumping the eeprom after writing with FT_PROG then comparing that with what pyftdi is attempting to write?

For instance, after programming with FT_PROG, use pyftdi to dump the current contents:

```python
from pyftdi.misc import hexdump
eeprom = FtdiEeprom()
eeprom.open('ftdi://ftdi:232:A50285BI/1')
print(hexdump(eeprom.data))
```

I have had a similar issue (`Error: Write to EEPROM failed @ 0x0e`). I'm using a F4232H. In my case, dumping the eeprom after using FT_PROG gave me this result:

```
000000   88 88 03 04 11 60 00 08 80 32 08 00 00 00 9a 0a   .....`...2......
000010   a4 0c b0 12 00 00 00 00 46 00 0a 03 41 00 53 00   ........F...A.S.
000020   33 00 46 00 0c 03 54 00 4D 00 5f 00 50 00 31 00   D.F...T.M._.P.1.
000030   12 03 46 00 54 00 35 00 31 00 37 00 31 00 46 00   ..F.T.5.1.7.1.F.
000040   47 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   G...............
000050   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
000060   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
000070   00 00 00 00 00 00 00 00 00 00 00 00 00 00 81 62   ...............b
000080   88 88 03 04 11 60 00 08 80 32 08 00 00 00 9a 0a   .....`...2......
000010   a4 0c b0 12 00 00 00 00 46 00 0a 03 41 00 53 00   ........F...A.S.
000020   33 00 46 00 0c 03 54 00 4D 00 5f 00 50 00 31 00   D.F...T.M._.P.1.
0000b0   12 03 46 00 54 00 35 00 31 00 37 00 31 00 46 00   ..F.T.5.1.7.1.F.
0000c0   47 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   G...............
0000d0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0000e0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0000f0   00 00 00 00 00 00 00 00 00 00 00 00 00 00 81 62   ...............b
```

The interesting thing to notice here is that the ftdi config is duplicated across the two halves of the eeprom. Is yours the same? Perhaps this is some setting in FT_PROG I clicked that configures the eeprom to function this way? I dont know...

Regardless, pyftdi does not produce a 'mirrored' configuration like this. Instead it places some pieces in the first sector (properties) and other pieces (var_strings) in the second sector. I [have a branch](https://github.com/len0rd/pyftdi/tree/feature/len0rd/eeprom_mirror) which allows you to specify if youd like to use a mirrored setup on construction. Setting `mirror=True` will result in the eeprom being written in a similar fashion to the above dump. Doing this solved the write errors for me, and gave me a valid configuration that works between power cycles. Here's what I ran:

```python
from pyftdi.misc import hexdump
eeprom = FtdiEeprom(mirror=True) # here is where you specify you want the eeprom to be mirrored
eeprom.open('ftdi://ftdi:232:A50285BI/1 ')
# I also had to add these as my ftdi eeprom comes from the factory uninitialized:
# eeprom.erase(erase_byte=0x00)
# eeprom.initialize()
eeprom.set_product_name('NANOU1')
eeprom.commit(dry_run=False)
print(hexdump(eeprom.data))
```
